Sources/Subprocess/Configuration.swift

```diff
@@ -815,34 +594,83 @@ internal struct TrackedFileDescriptor {
         self.closeWhenDone = closeWhenDone
     }
 
-    internal func safelyClose() throws {
+    #if os(Windows)
+    consuming func consumeDiskIO() -> FileDescriptor {
+        let result = self.fileDescriptor
+        // Transfer the ownership out and therefor
+        // don't perform close on deinit
+        self.closeWhenDone = false
+        return result
+    }
+    #endif
+
+    internal mutating func safelyClose() throws {
         guard self.closeWhenDone else {
             return
         }
+        closeWhenDone = false
 
         do {
             try fileDescriptor.close()
         } catch {
             guard let errno: Errno = error as? Errno else {
                 throw error
             }
-            if errno != .badFileDescriptor {
-                throw errno
+            // Getting `.badFileDescriptor` suggests that the file descriptor
+            // might have been closed unexpectedly. This can pose security risks
+            // if another part of the code inadvertently reuses the same file descriptor
+            // number. This problem is especially concerning on Unix systems due to POSIX’s
+            // guarantee of using the lowest available file descriptor number, making reuse
+            // more probable. We use `preconditionFailure` upon receiving `.badFileDescriptor`
+            // to prevent accidentally closing a different file descriptor.
+            guard errno != .badFileDescriptor else {
+                preconditionFailure(
```

Comment:
Should this be `fatalError`? The string of `preconditionFailure` is not retained in release builds, which will make the cause of crashes here, harder to identify.